src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/dormant/DormantEventsAboutToSubmitHandler.java

```diff
@@ -46,6 +66,44 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
         caseData.setDirectionDueDate(null);
         callback.getCaseDetailsBefore().ifPresent(f -> caseData.setPreviousState(f.getState()));
 
+        String caseID = callback.getCaseDetails().getCaseData().getCcdCaseId();
+        if (eligibleForHearingsCancel.test(callback)) {
+
+            CancellationReason cancellationReason = null;
+            switch (callback.getEvent()) {
+                case CONFIRM_LAPSED:
+                case HMCTS_LAPSE_CASE:
+                case ADMIN_SEND_TO_DORMANT_APPEAL_STATE:
+                case LAPSED_REVISED:
+                    cancellationReason = CancellationReason.LAPSED;
+                    break;
+                case WITHDRAWN:
+                    cancellationReason = CancellationReason.WITHDRAWN;
+                    break;
+                default: log.info("({}) Event is not handled for cancellation reason", callback.getEvent());
```

Comment:
Could we also log the case id along with the eventname for visibility in the logs